apps/dashboard/app/models/configuration_property.rb

```diff
@@ -0,0 +1,74 @@
+class ConfigurationProperty
+
+  def self.with_boolean_mapper(name:, default_value: nil, read_from_env: false, env_names: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_names, BooleanMapper)
+  end
+
+  def self.with_integer_mapper(name:, default_value: nil, read_from_env: false, env_names: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_names, IntegerMapper)
+  end
+
+  def self.property(name:, default_value: nil, read_from_env: false, env_names: nil)
+    ConfigurationProperty.new(name, default_value, read_from_env, env_names, PassThroughMapper)
+  end
+
+  attr_reader :name, :default_value, :read_from_environment, :environment_names
+  alias_method :read_from_environment?, :read_from_environment
+
+  #
+  # Represents a profile based configuration property.
+  # These properties are used by the UserConfiguration class to manage all profile based configurations.
+  #
+  # The default environment name is created based on the property name: ["OOD_#{@name.to_s.upcase}"]
+  # The different mappers are used to transform string values from the environment to a type.
+  #
+  # @param name [symbol] name of the property.To be used by UserConfiguration to lookup a value in the config object.
+  # @param default_value [any] default value for the property. To be used by UserConfiguration when no value is defined.
+  # @param read_from_env [boolean] if true, UserConfiguration will use the ENV to lookup a value.
+  # @param env_names [array] list of environment names to lookup a value for this property in the ENV object.
+  # @param mapper [Mapper Class] Mapper to transform strings from the environment to the property type.
+  #
+  def initialize(name, default_value, read_from_env, env_names, mapper)
+    @name = name.to_sym
+    @default_value = default_value
+    @read_from_environment = read_from_env
```

Comment:
```suggestion
    @read_from_environment = !!read_from_env
```

Let's cast this to a boolean, especially if we're relying on it `read_from_environment?`. I think `truthy` is probably enough, but I do kind of value safety.

Reply:
Will update